tree/ntuple/v7/src/RMiniFile.cxx

```diff
@@ -1190,6 +1190,16 @@ ROOT::Experimental::Internal::RNTupleFileWriter::Append(std::string_view ntupleN
    return writer;
 }
 
+void ROOT::Experimental::Internal::RNTupleFileWriter::Seek(std::uint64_t offset)
+{
+   if (!fFileSimple)
+      throw RException(R__FAIL("invalid attempt to seek non-simple writer"));
+
+   fFileSimple.fFilePos = offset;
```

Comment:
- should we do bounds checking on the offset?
- should we return a RResult<void> instead of throwing?

Reply:
I'm not sure, which bounds should we compare `offset` to? The idea is that external clients can use the method to skip over potentially a big amount of data, which the `RNTupleFileWriter` doesn't know about.

Regarding `RResult`, my understanding was that we use it for error conditions because of user input. That's not really the case here, it's a logic error in the program to call `Seek` on a non-simple writer. A good analogy would be passing a non-matching entry to `RNTupleWriter::Fill()` or `RNTupleReader::LoadEntry` - in both cases we also throw an exception.